drivers/flash/spi_nand.c

```diff
@@ -0,0 +1,1024 @@
+				goto out1;
+			}
+
+			release_device(dev);
+		}
+
+		if (onfi_table[167] & 0x01) {
+			data->read_recovery = true;
+		} else {
+			data->read_recovery = false;
+		}
+
+		if (onfi_table[168] & 0x02) {
+			data->continuous_read = true;
+
+			ret = spi_nand_conti_read_enable(dev, true);
+			if (ret != 0) {
+				LOG_ERR("SPI NAND Set continuous read enable Failed: %d", ret);
+				return ret;
+			}
+		} else {
+			data->continuous_read = false;
+		}
+	} else {
+		LOG_ERR("ONFI table not found");
+		return 0;
+	}
+
+	acquire_device(dev);
+
+	ret = spi_nand_get_feature(dev, SPI_NAND_FEA_ADDR_CONF_B0, &secur_reg);
+	if (ret != 0) {
+		LOG_ERR("get feature failed: %d", ret);
+		goto out1;
+	}
+
+	secur_reg &= ~SPINAND_SECURE_BIT_OTP_EN;
+	ret = spi_nand_set_feature(dev, SPI_NAND_FEA_ADDR_CONF_B0, secur_reg);
+	if (ret != 0) {
+		LOG_ERR("set feature failed: %d", ret);
+		goto out1;
+	}
+
+	ret = spi_nand_get_feature(dev, SPI_NAND_FEA_ADDR_CONF_B0, &secur_reg);
+	if (ret != 0) {
+		LOG_ERR("get feature failed: %d", ret);
+		goto out1;
+	}
+
+	if (secur_reg & SPINAND_SECURE_BIT_OTP_EN) {
+		LOG_ERR("Disable OTP failed: %d\n", secur_reg);
+		ret = -EINVAL;
+	}
+
+out1:
+	release_device(dev);
+	return ret;
+}
+
+/**
+ * @brief Configure the flash
+ *
+ * @param dev The flash device structure
+ * @param info The flash info structure
+ * @return 0 on success, negative errno code otherwise
+ */
+static int spi_nand_configure(const struct device *dev)
+{
+	const struct spi_nand_config *cfg = dev->config;
+
+	uint8_t reg = 0;
+	int rc;
+
+	/* Validate bus and CS is ready */
+	if (!spi_is_ready_dt(&cfg->spi)) {
+		return -ENODEV;
+	}
+
+	rc = spi_nand_check_id(dev);
+	if (rc != 0) {
+		LOG_ERR("Check ID failed: ");
+		return -ENODEV;
+	}
+
+	/* Check for block protect bits that need to be cleared. */
+	acquire_device(dev);
+
+	rc = spi_nand_get_feature(dev, SPI_NAND_FEA_ADDR_BLOCK_PROT, &reg);
+
+	if (rc != 0) {
+		LOG_ERR("get feature failed: %d", rc);
+		goto out;
+	}
+
+	/* Only clear if GET_FEATURE worked and something's set. */
+	if (reg & SPINAND_BLOCK_PROT_BIT_BP_MASK) {
+		reg = 0;
+		rc = spi_nand_set_feature(dev, SPI_NAND_FEA_ADDR_BLOCK_PROT, reg);
+		if (rc != 0) {
+			LOG_ERR("set feature failed: %d", rc);
+		}
+	}
+
+out:
+	release_device(dev);
+
+	if (rc != 0) {
+		return rc;
+	}
+
+	rc = spi_nand_read_otp_onfi(dev);
+	if (rc != 0) {
+		LOG_ERR("SFDP read failed: %d", rc);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Initialize and configure the flash
+ *
+ * @param name The flash name
+ * @return 0 on success, negative errno code otherwise
+ */
+static int spi_nand_init(const struct device *dev)
+{
+	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
+		struct spi_nand_data *const driver_data = dev->data;
+
+		k_sem_init(&driver_data->sem, 1, K_SEM_MAX_LIMIT);
+	}
+
+	return spi_nand_configure(dev);
+}
+
+static const struct flash_parameters flash_nand_parameters = {
+	.write_block_size = DT_INST_PROP(0, sub_page_size),
```

Comment:
This is same thing as assigned to spi_nand_config.sub_page_size, is this correct?